src/Process/Build/SetErrorReportDirNestingLevel.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Process\Build;
+
+use Magento\MagentoCloud\Config\Stage\BuildInterface;
+use Magento\MagentoCloud\Filesystem\ConfigFileList;
+use Magento\MagentoCloud\Filesystem\Driver\File;
+use Magento\MagentoCloud\Process\ProcessException;
+use Magento\MagentoCloud\Process\ProcessInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class SetErrorReportDirNestingLevel implements ProcessInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var ConfigFileList
+     */
+    private $configFileList;
+
+    /**
+     * @var BuildInterface
+     */
+    private $stageConfig;
+
+    /**
+     * @var File
+     */
+    private $file;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param ConfigFileList $configFileList
+     * @param BuildInterface $stageConfig
+     * @param File $file
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        ConfigFileList $configFileList,
+        BuildInterface $stageConfig,
+        File $file
+    ) {
+        $this->logger = $logger;
+        $this->configFileList = $configFileList;
+        $this->stageConfig = $stageConfig;
+        $this->file = $file;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        $this->logger->info('Configuring directory nesting for saving error reports');
+        try {
+            $errorReportConfigFile = $this->configFileList->getErrorReportConfig();
+            if ($this->file->isExists($errorReportConfigFile)) {
```

Comment:
Is it possible way that customers have this file, but without this variable?

Reply:
I see that `pub/errors/local.xml.sample` is not new file and it is quite possible that users already have it without `.sample`
So, need to parse this xml and check variable on existence

Reply:
Btw, seems like customer cannot override this value in `.env.yaml` if file already exists. Or you override it somewhere else?

Reply:
@NadiyaS 
Following a conversation with @YPyltiai, the following cases were agreed:

1. Customer does not have file:
1.1 No ERROR_REPORT_DIR_NESTING_LEVEL variable in env.yaml - we create a file and set default value 1
1.2 Variable exists in env.yaml - we create a file and set a custom value from ERROR_REPORT_DIR_NESTING_LEVEL var

2. If the client has a file, then we do not modify it.
At the deployment stage, we notify the client that a directories nesting level of directories for error reporting has not been set. And we propose to fix it.

Reply:
@YPyltiai not applying configuration value from .magento.env.yaml goes against the behavior we have for other configurations. How do we plan to make it clear to customers? (let's say that nobody will memorise every exception for every configuration's behavior. Better to say that all what you set in .env.yaml will override existed configs - it is not clear and more expected for everyone).

Reply:
@NadiyaS case # 2 is very unlikely to happen - this logic just protects us from situations where we override something we shouldn't touch. It's very unlikely that someone will have that file, so in most cases behavior will be as described in # 1 - it will use value from env.yaml or follow default behavior otherwise. I think it's not the first variable with overrides we have - it just needs to be well documented.